pkg/agent/manager/cache/cache.go

```diff
@@ -228,6 +228,10 @@ func (c *Cache) Update(update *Update, checkSVID func(*common.RegistrationEntry,
 	fedRem, fedRemDone := allocStringSet()
 	defer fedRemDone()
 
+	var deleteCounter float32
```

Comment:
Rather than have much of this telemetry behavior leak into the caching logic, i'd prefer to see an API like:

```
metricsHelper := newCacheMetricsHelper(c.metrics)
defer metricsHelper.Finish()

...
if _, ok := update.RegistrationEntries[id]; !ok {
    ...
    metricsHelper.IncCacheDelete()
}

...
if existingEntry != nil {
    metricsHelper.IncCacheUpdate()
} else {
    metricsHelper.IncCacheCreate()
}
```

Reply:
removed this part.